validator/valnode/valnode.go

```diff
@@ -0,0 +1,126 @@
+)
+
+type WasmConfig struct {
+	RootPath string `koanf:"root-path"`
+}
+
+func WasmConfigAddOptions(prefix string, f *flag.FlagSet) {
+	f.String(prefix+".root-path", DefaultWasmConfig.RootPath, "path to machine folders, each containing wasm files (replay.wasm, wasi_stub.wasm, soft-float.wasm, go_stub.wasm, host_io.wasm, brotli.wasm")
+}
+
+var DefaultWasmConfig = WasmConfig{
+	RootPath: "",
+}
+
+type Config struct {
+	UseJit     bool                               `koanf:"use-jit"`
+	ApiAuth    bool                               `koanf:"api-auth"`
+	ApiPublic  bool                               `koanf:"api-public"`
+	Arbitrator server_arb.ArbitratorSpawnerConfig `koanf:"arbitrator" reload:"hot"`
+	Jit        server_jit.JitSpawnerConfig        `koanf:"jit" reload:"hot"`
+	Wasm       WasmConfig                         `koanf:"wasm"`
+}
+
+type ValidationConfigFetcher func() *Config
+
+var DefaultValidationConfig = Config{
+	Jit:        server_jit.DefaultJitSpawnerConfig,
+	ApiAuth:    true,
+	ApiPublic:  false,
+	Arbitrator: server_arb.DefaultArbitratorSpawnerConfig,
+	Wasm:       DefaultWasmConfig,
+}
+
+var TestValidationConfig = Config{
+	Jit:        server_jit.DefaultJitSpawnerConfig,
+	ApiAuth:    false,
+	ApiPublic:  true,
+	Arbitrator: server_arb.DefaultArbitratorSpawnerConfig,
+	Wasm:       DefaultWasmConfig,
+}
+
+func ValidationConfigAddOptions(prefix string, f *flag.FlagSet) {
+	f.Bool(prefix+".use-jit", DefaultValidationConfig.UseJit, "use jit for validation")
+	f.Bool(prefix+".api-auth", DefaultValidationConfig.ApiAuth, "validate is a public API")
+	f.Bool(prefix+".api-public", DefaultValidationConfig.ApiPublic, "validate is an authenticated API")
+	server_arb.ArbitratorSpawnerConfigAddOptions(prefix+".arbitrator", f)
+	server_jit.JitSpawnerConfigAddOptions(prefix+".jit", f)
+	WasmConfigAddOptions(prefix+".wasm", f)
+}
+
+type ValidationNode struct {
+	config     ValidationConfigFetcher
+	arbSpawner *server_arb.ArbitratorSpawner
+	jitSpawner *server_jit.JitSpawner
+}
+
+func CreateValidationNode(configFetcher ValidationConfigFetcher, stack *node.Node, fatalErrChan chan error) (*ValidationNode, error) {
+	config := configFetcher()
+	locator, err := server_common.NewMachineLocator(config.Wasm.RootPath)
+	if err != nil {
+		return nil, err
+	}
+	arbConfigFetcher := func() *server_arb.ArbitratorSpawnerConfig {
+		return &configFetcher().Arbitrator
+	}
+	arbSpawner, err := server_arb.NewArbitratorSpawner(locator, arbConfigFetcher)
+	if err != nil {
+		return nil, err
+	}
+	node.DefaultAuthModules = []string{server_api.Namespace}
+	var serverAPI *server_api.ExecServerAPI
+	var jitSpawner *server_jit.JitSpawner
+	if config.UseJit {
+		jitConfigFetcher := func() *server_jit.JitSpawnerConfig { return &configFetcher().Jit }
+		var err error
+		jitSpawner, err = server_jit.NewJitSpawner(locator, jitConfigFetcher, fatalErrChan)
+		if err != nil {
+			return nil, err
+		}
+		serverAPI = server_api.NewExecutionServerAPI(jitSpawner, arbSpawner)
```

Comment:
I'm not sure if I commented on this somewhere else, so apologies if this is a duplicate comment, but I think we want to support doing both arbitrator and jit validation on a validator for our makenodes validators. The jit validation is cheap in comparison to arbitrator and ensures other jit validators will agree with the assertion.